### PR TITLE
fix: upload templates github action

### PR DESCRIPTION
**Pull request by TBonnin**
Templates relying on zod are failing to be uploaded. ex: https://github.com/NangoHQ/nango/actions/runs/10489880648/job/29055453274

This commit:
- install zod before compiling
- add a way to manually trigger uploading all templates

@bodinsamuel can you think of a way to safely test this? 

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: .github/workflows/integration-template-upload.yaml [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
 on:
+    workflow_dispatch:
     push:
         branches:
             - master
-    pull_request:
-        branches:
-            - master
-        types:
-            - closed
```
Comment (TBonnin) [inline .github/workflows/integration-template-upload.yaml:12]: I think the pull request trigger is redundant. We want to upload when a commit is pushed to master, that's it.

**Comment by khaliqgant** [comment]
https://github.com/NangoHQ/nango/blob/master/.github/workflows/sync-integration-templates.yaml

**Comment by TBonnin** [comment]
> https://github.com/NangoHQ/nango/blob/master/.github/workflows/sync-integration-templates.yaml

Missed this one 😄. Not sure we need 2 actions doing almost the same thing.

**Comment by bodinsamuel** [comment]
> https://github.com/NangoHQ/nango/blob/master/.github/workflows/sync-integration-templates.yaml

Arg, my bad for leading you into this @TBonnin 
Might still be a good idea to have one way to push templates though

File: .github/workflows/integration-template-upload.yaml [hunk 2]
```diff
@@ -56,10 +49,11 @@ jobs:
                     cp -r $dir/* /tmp/nango-temp/nango-integrations/$integration/
 
                     cd /tmp/nango-temp/nango-integrations
+                    nango install zod
```
Comment (bodinsamuel) [inline .github/workflows/integration-template-upload.yaml]: npm install zod?

**Review by bodinsamuel — APPROVED** [review]
Looks good, github actions are notoriously painful to debug so I guess you can either trigger manually on your branch or test in master. Happy to review another PR if any bug arises 👍🏻

File: .github/workflows/integration-template-upload.yaml (hunk 2 above)
Comment (bodinsamuel) [reply .github/workflows/integration-template-upload.yaml]: now that I think about it we might also need `soap` and botbuilder

**Comment by TBonnin** [comment]
> Looks good, github actions are notoriously painful to debug so I guess you can either trigger manually on your branch or test in master. Happy to review another PR if any bug arises 👍🏻

`on:workflow_dispatch` is not enabled until this is merged to master unfortunately. 

Edit: giving me an idea, let me do a simple PR that is just adding on:workflow_dispatch. Then I will be able to test the script changes

**Comment by TBonnin** [comment]
Testing the manual upload to sync all templates. Merging now to ensure the merge based logic is also working

**Pull request merged**